lib/mozart_fetcher/fetcher.ex

```diff
@@ -11,20 +11,35 @@ defmodule MozartFetcher.Fetcher do
 
   def process(components) do
     ExMetrics.timeframe "function.timing.fetcher.process" do
-      max_timeout =
-        TimeoutParser.max(components) +
-          Application.get_env(:mozart_fetcher, :additional_task_await_timeout)
+      max_timeout = TimeoutParser.max(components)
+      stream_opts = [timeout: max_timeout, on_timeout: :kill_task, max_concurrency: 40]
 
       components
       |> Enum.with_index()
-      |> Enum.map(&Task.async(fn -> Component.fetch(&1) end))
-      |> Enum.map(&Task.await(&1, max_timeout))
-      |> decorate_response
+      |> Task.async_stream(&Component.fetch/1, stream_opts)
+      |> Enum.to_list()
+      |> extract_successful()
+      |> decorate_response()
       |> Jason.encode!()
     end
   end
 
   defp decorate_response(envelopes) do
     %{components: envelopes}
   end
+
+  # returns the successful one (including 500 etc) and logs
+  # the tasks which have not completed.
+  # At the moment it's in form  of `[exit: :timeout]` so hard to
+  # easily log which component has failed.
+  defp extract_successful(responses) do
+    {oks, errors} = Enum.split_with(responses, fn {k, _v} -> k == :ok end)
+
+    errors |> Enum.each(&log(&1))
```

Comment:
Do we need to do an each here if it's only errors as we could pass all and log all at once?